streaming/base/util.py

```diff
@@ -253,6 +254,50 @@ def merge_index(*args: Any, **kwargs: Any):
     raise ValueError(f'Invalid arguments to merge_index: {args}, {kwargs}')
 
 
+def _download_url(url_info):
+    """Download a file given URL information."""
+    from streaming.base.storage.download import download_file
+    src, dest, download_timeout = url_info
+    try:
+        download_file(src, dest, download_timeout)
+    except Exception as ex:
+        return f'Failed to download index.json: {src} to {dest}: {str(ex)}', ex
+    return dest, None
+
+
+def _merge_partition_indices(partition_indices):
+    """Function to be executed by each process to merge a subset of partition indices."""
+    shards = []
+    for partition_index in partition_indices:
+        p = Path(partition_index)
+        with open(partition_index, 'r') as f:
+            obj = json.load(f)
+        for shard in obj['shards']:
+            for key in ('raw_data', 'zip_data', 'raw_meta', 'zip_meta'):
```

Comment:
we really ought to make this a Shard method, which is subject to inheritance and so on

this code won't work for parquet shards :/

Reply:
Any specific suggestion how to deal with this?